Implemented more common check_mk agent sections

jens-maus opened the pull request:
This PR implements:

- using `timeout` wrapper script to secure `tcp_conn_stats` and `chrony`/`ntp` executions which could hang otherwise
- implements `<<<nfsmounts>>>` and `<<<cifsmounts>>>` sections if `/bin/stat` exists and waitmax is there.
- revised `<<<lnx_thermal>>>` check_mk output to add all existing thermal zones actually existing and not just thermal_zone0
- uses `/VERSION` rather than `/boot/VERSION` to query the firmware version if `/VERSION` exists (which is the preferred way for CCU3/RaspberryMatic).

jens-maus committed:
added ps and ethtool aware check_mk output. Please note that the output
of ps is somewhat limited to the capabilities to the ps command of
busybox for which conversion routines might be a solution in future. But
for the moment this should be enough to actually monitor if a certain
process is running or not.

jens-maus committed:
made 'ps' call compatible to old homematic ccu2 busybox ps version.

jens-maus committed:
fixed ps call for CCU2 once more.

jens-maus committed:
Merge remote-tracking branch 'upstream/master'

jens-maus committed:
check for /VERSION first before using /boot/VERSION (improves CCU3/RaspberryMatic
compatibility)

jens-maus committed:
added more lnx_thermal information so that the thermal zone of a GPU is
also added to the check_mk output as well as the trip_point limits
defined by the hardware.

jens-maus committed:
added "waitmax" binary for CCU2 and CCU3 platform and propagated its use
to the check_mk agent execution. Also implemented nfsmounts and
cifsmounts outputs using waitmax.

alexreinert commented:
Thanks for the PR.
The inclusion of waitmax violates with the GPLv2, because of that I cannot merge this PR.

jens-maus commented:
Then feel free to remove the waitmax use from the PR.

Or reconsider changing the license of your Addon to GPLv2 itself or add the waitmax sources to your repository to show that you haven't changed anything.

jens-maus committed:
implemented additional chrony NTP daemon check because upcoming RaspberryMatic
versions will use chrony instead of NTPd.

jens-maus committed:
changed waitmax to use the coreutils `timeout` command instead which
should provide the same functionality but is licensed under GPL3 which
is compatible to Apache2.

jens-maus commented:
@alexreinert Please note that with the latest commit I retired `waitmax` and introduced the coreutils `timeout` command which provides essentially the same functionality but is licensed under the GPL3 license which seem to be compatible to the Apache2.0 license (see https://www.apache.org/licenses/GPL-compatibility.html) this project is licensed.

alexreinert commented:
The linked page states, that this direction is not possible.
```
However, GPLv3 software cannot be included in Apache projects.
```
Only GPL license, which would be compatible in that way, would be the LGPL.

And I will not include binaries in my repositories, where I do not know, how they are build. (Because of security AND because of support)

jens-maus committed:
introduced a wrapper script to either directly call an existing
'timeout' command (e.g. on RaspberryMatic) or to use shell syntax to
replicate the same functionality just using shell functionality. This
should hopefully solve any licensing issues between GPL and Apache2
license under which this check_mk_agent is licensed.

jens-maus commented:
Woooh. Ok, I definitely overlooked that final sentence. Never mind, I just introduced a replacement wrapper script in my last commit (see fcc773b) and explicitly licensed it under Apache2.0 for your convenience. This hopefully should solve your license concerns while at the same time provide a reliable `timeout` functionality for all supported CCU platforms.

alexreinert commented:
Looks better now, thanks.
I will test all changes on the supported platforms, hopefully I can do this 'til the weekend.

jens-maus committed:
fixed "timeout" wrapper script to correctly replicate the BusyBox
'timeout' command syntax where the time to wait is supplied via a second
argument rather than a "-t" option.

jens-maus committed:
removed excessive syslog output uses to stop flooding with just
informational messages.

jens-maus commented:
any progress here? I would like to finished this PR and get my fork cleanup up accordingly. Would be nice to see this PR finally integrated. Here it is working stable since then
